documents/model.go

```diff
@@ -25,3 +31,134 @@ type Model interface {
 	// CreateProofs creates precise-proofs for given fields
 	CreateProofs(fields []string) (coreDoc *coredocumentpb.CoreDocument, proofs []*proofspb.Proof, err error)
 }
+
+// CoreDocumentModel contains methods which handle all interactions mutating or reading from a core document
+// Access to a core document should always go through this model
+type CoreDocumentModel struct {
+	document *coredocumentpb.CoreDocument
+}
+
+// NewCoreDocModel returns a new CoreDocumentModel
+// Note: collaborators and salts are to be filled by the caller
+func NewCoreDocModel() *CoreDocumentModel {
+	id := utils.RandomSlice(32)
+	cd := &coredocumentpb.CoreDocument{
+		DocumentIdentifier: id,
+		CurrentVersion:     id,
+		NextVersion:        utils.RandomSlice(32),
+	}
+	return &CoreDocumentModel{
+		cd,
+	}
+}
+
+// GetDocument returns the coredocument from the CoreDocumentModel
+func (m *CoreDocumentModel) GetDocument() (*coredocumentpb.CoreDocument, error) {
+	cd := m.document
+	if cd == nil {
+		return nil, errors.New("error getting document in CoreDocModel")
+	}
+	return m.document, nil
+}
+
+// SetDocument set the coredocument in the CoreDocumentModel to an updated coredocument
+func (m *CoreDocumentModel) SetDocument(ncd *coredocumentpb.CoreDocument) *CoreDocumentModel {
```

Comment:
Let's not expose the CoreDocument through setters, all mutations on the CD should be done through CoreDocModel